// nestjs-form-data/dist/decorators/validation/has-mime-type.validator.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.HasMimeType = void 0;
const class_validator_1 = require("class-validator");
const is_file_validator_1 = require("./is-file.validator");
const toArray_1 = require("../../helpers/toArray");
function HasMimeType(allowedMimeTypes, strictSource, validationOptions) {
    return class_validator_1.ValidateBy({
        name: 'HasMimeType',
        constraints: [allowedMimeTypes],
        validator: {
            validate(value, args) {
                const allowedMimeTypes = toArray_1.toArray(args.constraints[0]) || [];
                const strictSource = (typeof args.constraints[1] === 'string')
                    ? args.constraints[1]
                    : undefined;
                if (is_file_validator_1.isFile(value)) {
                    const mimeWithSource = value.mimeTypeWithSource;
                    return allowedMimeTypes.includes(mimeWithSource.value) && (!strictSource || strictSource === mimeWithSource.source);
                }
                return false;
            },
            defaultMessage(validationArguments) {
                const allowedMimeTypes = toArray_1.toArray(validationArguments.constraints[0]) || [];
                return `File must be of one of the types ${allowedMimeTypes.join(', ')}`;
            },
        },
    }, validationOptions || ((typeof strictSource === 'object') ? strictSource : null));
}
exports.HasMimeType = HasMimeType;
